sqlglot/generator.py

```diff
@@ -1520,19 +1521,23 @@ def withdataproperty_sql(self, expression: exp.WithDataProperty) -> str:
         return f"{data_sql}{statistics_sql}"
 
     def withsystemversioningproperty_sql(self, expression: exp.WithSystemVersioningProperty) -> str:
-        sql = "WITH(SYSTEM_VERSIONING=ON"
-
-        if expression.this:
-            history_table = self.sql(expression, "this")
-            sql = f"{sql}(HISTORY_TABLE={history_table}"
+        this = self.sql(expression, "this")
+        this = f"HISTORY_TABLE={this}" if this else ""
+        data_consistency = self.sql(expression, "data_consistency")
+        data_consistency = f"DATA_CONSISTENCY_CHECK={data_consistency}" if data_consistency else ""
+        retention_period = self.sql(expression, "retention_period")
+        retention_period = (
+            f"HISTORY_RETENTION_PERIOD={retention_period}" if retention_period else ""
+        )
 
-            if expression.expression:
-                data_consistency_check = self.sql(expression, "expression")
-                sql = f"{sql}, DATA_CONSISTENCY_CHECK={data_consistency_check}"
+        if this:
+            on_sql = self.func("ON", this, data_consistency or None, retention_period or None)
```

Comment:
We can get rid of the `or None`s and just move them at the definition site `else` branch